src/main/resources/templates/user/cartOverview.html

```diff
@@ -0,0 +1,130 @@
+                            <a th:href="@{/users/cart/details/minus_one/{id}(id=${product.id})}"
+                               style="text-decoration: none;">
+                                <label class="cart-product-item-content cart-detail-href">
+                                    <em class="material-icons" style="font-size: 20px; padding-top: 9px">remove_circle_outline</em>
+                                </label>
+                            </a>
+                        </span>
+                        <!-- if above is not true show this link that just links to the modal by the css class name -->
+                        <span th:unless="${product.totalInCart > 1}">
+                            <a style="text-decoration: none;">
+                                <label class="cart-product-item-content cart-detail-href ">
+                                    <em th:id="${'minusid'+index.count}" class="material-icons show-modal"
+                                        style="font-size: 20px; padding-top: 9px">remove_circle_outline</em>
+                                </label>
+                            </a>
+                        </span>
+                    </div>
+                    <div class="cart-product-item-flex-product-count">
+                        <label class="cart-product-item-content"
+                               th:text="${product.totalInCart}">99</label>
+                    </div>
+                    <div class="cart-product-item-flex-product-plus">
+                        <a th:href="@{/users/cart/details/plus_one/{id}(id=${product.id})}"
+                           style="text-decoration: none;">
+                            <label class="cart-product-item-content cart-detail-href">
+                                <em class="material-icons"
+                                    style="font-size: 20px; padding-top: 9px">add_circle_outline</em>
+                            </label>
+                        </a>
+                    </div>
+                    <div class="cart-product-item-flex-product-total">
+                        <label class="cart-product-item-title"
+                               th:text="'€ ' + ${product.totalInCart*product.price}">€
+                            999.99</label>
+                    </div>
+                    <div class="cart-product-item-flex-product-delete">
+                        <a th:href="@{/users/cart/details/delete/{id}(id=${product.id})}"
+                           style="text-decoration: none;">
+                            <label class="cart-product-item-content cart-detail-href">
+                                <em class="material-icons"
+                                    style="font-size: 26px; padding-top: 9px">delete_outline</em>
+                            </label>
+                        </a>
+                    </div>
+                    <dialog th:if="${product.totalInCart == 1}" th:id="${'dialogid'+index.count}" class="mdl-dialog">
```

Comment:
this dialog is your popup. It can be lowered down inside the template, thus making it clear that it isn't a view when reading the code. Advice: place it right above <script> and place <!--popup window--> above <dialog>

Reply:
it should be inside the `<li>` because it's specific to each `<li>` generated by the `th:each` and it has the specific link for that iteration on the remove-button, it aint pretty but it's supposed to be there

Reply:
figured we'd removed that logic by setting the id's ourselves. But you're right! Forget about the links